### ISSUE TITLE
On disconnect from twilio video call, webcam light still remains active.

### ISSUE THREAD

**Issue by mayurgujrathi** [twilio/video-quickstart-js#102]
I have developed video call application using angular 8 where participant joins in a video call and can see his local video turning webcam light ON. But when leaving or video call closing tab webcam light is not getting turned OFF. Library used for twilio video call is 2.0.0
I dont know what i am doing wrong.

On close button click I am calling a function in typescript as follows.

closeTimeEditDialog() {
   
    this.room.localParticipant.tracks.forEach(publication => {
        const attachedElements = publication.track.detach();
        publication.track.stop();
            attachedElements.forEach(element => element.remove());
        });
        this.room.disconnect();
        this.messageService.showVideocallScreen = false;
}

    room.once('disconnected', () => {
        this.room.localParticipant.tracks.forEach(publication => {
        publication.track.stop();
        const attachedElements = publication.track.detach();
        attachedElements.forEach(element => element.stop());
    });

});

**Comment by manjeshbhargav** [comment]
Hi @mayurgujrathi ,

Sorry for the late response. How are you capturing local media? Are you calling `createLocalTracks()` and passing the resulting `tracks` to ConnectOptions, or are you passing `audio` and `video` capture constraints directly to ConnectOptions? In the first case, you would have to call `stop()` on each Track. In the second case, the SDK does it for you when you leave the Room.

Is there a way you can share the full source code with me. I'd be happy to take a look.

Thanks,

Manjesh Malavalli
JSDK Team
